web/frontend/src/api/navigation.ts

```diff
@@ -41,7 +41,7 @@ export const setWaypoint = (lat: number, lng: number, name: string) => {
   rcLogConditionally(request);
 
   return new Promise((resolve, reject) => {
-    client.current.navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) => {
+    client.navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) => {
```

Comment:
Also not really a part of this PR, but thoughts on overall architecture of the TypeScript SDK that this PR surfaces for me:

`client.navigationService.addWaypoint` is ringing some [law of demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) warning bells; `client` is a "complex" object that we have to reach "deep" inside to get work done, and we're passing it around via function arguments.

I think organizing the gRPC client code into different namespaces like `navigation` is useful, and it looks like some of these object-oriented structures are a little baked into our gRPC client generation code, which is a bit of a bummer.

I don't have any bright ideas here yet, since there seems to be an element of actual socket/session connectedness wrapped up in the `client`'s state. But I wonder if there's a world where we import functions like `addWaypoint` directly from the TypeScript SDK and pass in a simple object for parameters like the hostname, instead of having to create and hang onto these complex client objects in the calling code

Reply:
> client.navigationService.addWaypoint is ringing some [law of demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) warning bells; client is a "complex" object that we have to reach "deep" inside to get work done, and we're passing it around via function arguments.

Deeply agree. I left addressing this problem out of this current PR but have been thinking about the problem as well.

> I don't have any bright ideas here yet, since there seems to be an element of actual socket/session connectedness wrapped up in the client's state. But I wonder if there's a world where we import functions like addWaypoint directly from the TypeScript SDK and pass in a simple object for parameters like the hostname, instead of having to create and hang onto these complex client objects in the calling code.

Yep, I was considering the same. Or maybe in the future the typescript SDK should have a reference to a client that the user instantiates. This would make the assumption of one client per robot per app, which we don't currently make. most of my uncertainty with how to architect around the SDK has to do with my lack of feeling confident with how `client` should be used.

cc @maximpertsov

Reply:
> there seems to be an element of actual socket/session connectedness wrapped up in the client's state. 

Correct, that and the inherent structure of the auto-generated code are the main obstacles to achieving the interface you're describing, which I would also like to have. Short-term, I want to factor out the "connection" logic from `Client` and have users pass the former around - that might open a path to instantiating clients with either an open connection or a parameters object (i.e. URIs, secrets) to create a new connection under the hood.

Reply:
Also, just want to point out that this sort of deep method chain calling is only necessary for methods for which we haven't created wrappers yet - e.g. [raw nav service](https://ts.viam.dev/modules/navigationApi.html) vs [wrapped motion service](https://ts.viam.dev/classes/MotionClient.html) - or if you want to access the raw protobuf methods. 

All of your points about needing to passing around a complex "client" object still stand though.